File: dyte-objective-c/dyte-objective-c/ViewController.m
//
//  ViewController.m
//  dyte-objective-c
//
//  Created by Rohit Bhatia on 7/22/21.
//

#import "ViewController.h"
@import DyteSdk;

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}
- (IBAction)joinMeeting:(UIButton *)sender {
    DyteMeetingConfig *meetingConfig = [[DyteMeetingConfig alloc] init];
      meetingConfig.roomName = @"hazel-mile";
      meetingConfig.width = self.view.frame.size.width;
      meetingConfig.height = self.view.frame.size.height;
      DyteMeetingView *dyteView = [[DyteMeetingView alloc] initWithFrame:CGRectMake(0.0, 0.0, meetingConfig.width, meetingConfig.height)];
      dyteView.delegate = self;
      dyteView.tag = 100;
      [self.view addSubview:dyteView];
      [dyteView join:meetingConfig];
}

- (void)meetingEnded {
  dispatch_async(dispatch_get_main_queue(), ^{
    for (UIView *view in [self.view subviews])
    {
      if (view.tag == 100) {
        [view removeFromSuperview];
      }
    }
  });
}



@end

File: dyte-swift-example/dyte-swift-example/ViewController.swift
import UIKit
import DyteSdk;

class ViewController: UIViewController {
//    @IBOutlet weak var videoButton: UIButton?
    fileprivate var dyteView: DyteMeetingView?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = .red;


    }
    
//    @IBAction func joinMeeting(_ sender: Any) {
//            	let  config = DyteMeetingConfig();
//                config.roomName = "hazel-mile";
//                let dyteView = DyteMeetingView(frame: CGRect(x: 0, y: 0, width: self.view.bounds.size.width, height:self.view.bounds.size.height ))
//                dyteView.delegate = self
//                dyteView.tag  = 100
//                self.dyteView = dyteView;
//                self.view.addSubview(dyteView)
//                dyteView.join(config);
//    }
    
    fileprivate func onMeetingEnded() {
        if let viewWithTag = self.view.viewWithTag(100) {
            viewWithTag.removeFromSuperview()
        }
    }
}


extension ViewController: DyteMeetingViewDelegate {
    func meetingEnded() {
        DispatchQueue.main.async { [weak self] in
            self?.onMeetingEnded()
        }
    }
}

extension UIViewController {
    func hideKeyboardWhenTappedAround() {
        let tap = UITapGestureRecognizer(target: self, action: #selector(UIViewController.dismissKeyboard))
        tap.cancelsTouchesInView = false
        view.addGestureRecognizer(tap)
    }
    
    @objc func dismissKeyboard() {
        view.endEditing(true)
    }
}


